### DISCUSSION TITLE
Initialize object values from html value

### DISCUSSION THREAD

**Discussion by marcellelek** [alpinejs/alpine#610] [4. General]
Hi,
Thank you for this amazing project!
My question is, that is it possible to initialize the object values from the html input/select element's value.
I know it can be done via the x-init attribute, but my models are kinda big, so this should be not an option for me.

Codepen for demo:
https://codepen.io/marcellelek/pen/jOWLawX

My expected output would be that below the input, the text already contains "Test" on initialization.

Thank you for your answer.

**Comment by mblarsen** [comment]
Yeah there is not built-in way of doing that AFAIK. Here are a few ways. 

The **init-way** that you didn't want to:

```
<div x-data="objectData()" x-init="text = $ref.input.value">
    <input x-ref="input" value="Test"/>
</div>
```

**Passing $el** Some recent versions ago you now have access to `$el` in `x-data`:

```
<div x-data="objectData($el)">
    <input value="Test"/>
</div>

var objectData = function (el) {
  return {
    text: el.querySelector('input').value // first input child of your alpine component assuming you only have one
  }
}
```

**Backend render**, e.g. using Laravel Blade:

```
var objectData = function (el) {
  return {
    text: '{{ $value }}'
  }
}
```

**Reply by marcellelek** [reply]
Thank you for your reply, 
I went with my own, which kinda works.

Made a set property function:

 ```
setProperties: function ($el, _this) {
            Array.from($el.getElementsByTagName("input")).forEach(function (element, index) {
                var attributeNode = element.getAttributeNode('x-model');
                if (typeof (attributeNode) !== 'undefined' && attributeNode !== null) {
                    var propertyName = element.getAttributeNode('x-model').value;
                    var propertyValue = element.defaultValue

                    if (element.value)
                        propertyValue = element.defaultValue;

                    _this[propertyName] = propertyValue;
                }
            });
        }
```

Than using it with an init function:
```
        init: function ($el) {
            var _this = this;
            setProperties($el, _this);
        },
```
And in code i just call:

`  x-init="() => init($el)"`

Not the best solution, because i cannot work with hidden inputs this way, but, works for now.

**Reply by mblarsen** [reply]
Good that you found a solution :+1: 

```
 x-init="init($el)"
```

Doesn't this work? Don't think you need the anonymous function.

**Reply by marcellelek** [reply]
Ohh.. Yes it does. Thanks for that!

**Comment by SoftViruS** [comment]
`<div x-data="{ text: '' }"><input x-model="text = $el.value" value="test" /></div>`
